Public/Src/Tools/SymbolDaemon/VsoSymbolClient.cs

```diff
@@ -128,7 +128,12 @@ public async Task<Request> CreateAsync(CancellationToken token)
         public async Task<AddDebugEntryResult> AddFileAsync(SymbolFile symbolFile)
         {
             Contract.Requires(symbolFile.IsIndexed, "File has not been indexed.");
-            Contract.Requires(symbolFile.DebugEntries.Count > 0, "File contains no symbol data.");
+            
+            if (symbolFile.DebugEntries.Count == 0)
+            {
+                // early return if there are no debug entries.
+                return AddDebugEntryResult.NoSymbolData;
+            }

```

Comment:
If there are no symbols, we will still have the success result. That's fine, i guess. What about logging a warning on the BuildXL side, similarly to what I've been doing for collision? 

You can use "logmessagecommand" for that.